core-library/src/main/java/org/silverpeas/core/admin/user/model/UserDetail.java

```diff
@@ -724,27 +725,45 @@ public String getDurationOfCurrentSession() {
   }
 
   /**
-   * Indicates if the current user is in relation with, or invited by, a user represented by the
-   * given identifier.
-   * @param userId the identifier of the user which the current user is potentially in relation
-   * with or invited by.
-   * @return true if the current user is in relation with, or invited by, the user represented by
+   * Indicates if the current user is in relation with a user represented by the given identifier.
+   * @param userId the identifier of user which the current user is potentially in relation with.
+   * @return true if the current user is in relation with the user represented by
    * the given identifier, false otherwise.
    */
-  public boolean isInRelationWithOrInvitedBy(String userId) {
+  public boolean isInRelationWith(String userId) {
     RelationShipService relation = RelationShipService.get();
-    InvitationService invitation = InvitationService.get();
     try {
-      return relation.isInRelationShip(Integer.parseInt(userId), Integer.parseInt(getId())) ||
-          (invitation.getInvitation(Integer.parseInt(userId), Integer.parseInt(getId())) != null);
+      return relation.isInRelationShip(Integer.parseInt(userId), Integer.parseInt(getId()));
     } catch (Exception e) {
       SilverLogger.getLogger(this)
-          .warn("not possible to verify relation with or invitation of userId ''{0}'' ({0})",
+          .warn("not possible to verify relation with ''{0}'' ({0})",
               userId, e.getMessage());
     }
     return false;
   }
 
+  public Invitation getInvitationSentTo(String userId) {
+    return getInvitation(getId(), userId);
+  }
+
+  public Invitation getInvitationReceivedFrom(String userId) {
+    return getInvitation(userId, getId());
+  }
+
+  private Invitation getInvitation(String fromUserId, String toUserId) {
+    InvitationService invitationService = InvitationService.get();
+    Invitation invitation = null;
+    try {
+      invitation =
+          invitationService.getInvitation(Integer.parseInt(fromUserId), Integer.parseInt(toUserId));
+    } catch (Exception e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1166)